src/mdesktopaction.h

```diff
@@ -0,0 +1,46 @@
+/***************************************************************************
+** This file was derived from the MDesktopAction implementation in the
+** libmeegotouch library.
+**
+** Copyright (C) 2021 Chupligin Sergey (NeoChapay) <neochapay@gmail.com>
+**
+** This library is free software; you can redistribute it and/or
+** modify it under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation
+** and appearing in the file LICENSE.LGPL included in the packaging
+** of this file.
+**
+****************************************************************************/
+
+#ifndef MDESKTOPACTION_H
+#define MDESKTOPACTION_H
+
+#include <mlite-global.h>
+#include <QString>
+
+class MDesktopEntryPrivate;
+
+class MLITESHARED_EXPORT MDesktopAction
+{
+
+public:
+    MDesktopAction(const QString &fileName, const QString &action);
+    virtual ~MDesktopAction() {};
+
+    QString name() const;
+    QString nameUnlocalized() const;
+    QString getExec() const;
+    QString icon() const;
+    QString actionName() const {return m_action;}
```

Comment:
Better not to have code in headers if there's no specific reason if it's a public class.